realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -363,6 +373,13 @@ private void emitInitTableMethod(JavaWriter writer) throws IOException {
             writer.emitStatement("table.setIndex(table.getColumnIndex(\"%s\"))", fieldName);
         }
 
+        if (primaryKey != null) {
+            String fieldName = primaryKey.getSimpleName().toString();
+            writer.emitStatement("table.setPrimaryKey(\"%s\")", fieldName);
+        } else {
+            writer.emitStatement("table.setPrimaryKey(\"\")");
```

Comment:
Yes, as primary keys are saved in a meta table, we need to clear that if the primary key has been removed. This is just a "smart" migration.